File: blog-backend/src/main/java/com/example/blogbackend/services/article/ArticleService.java
package com.example.blogbackend.services.article;

import com.example.blogbackend.dtos.ArticleDTO;
import com.example.blogbackend.dtos.ArticleRequestDTO;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Set;
import java.util.UUID;

public interface ArticleService {
    Set<ArticleDTO> getAllArticles();

    Page<ArticleDTO> getAllArticlesPaged(int page, int size, String[] sort, String order);
    ArticleDTO getArticleById(UUID id);

    ArticleDTO createArticle(UUID userId, Set<UUID> categoryIds, ArticleRequestDTO articleRequestDTO);

    void deletePublication(UUID id);

    void addFavorite(UUID userId, UUID articleId);

    void removeFavorite(UUID userId, UUID articleId);

    List<ArticleDTO> getFavoriteArticlesByUser(UUID userId);
}

File: blog-front/src/api/articleService.js
import axiosInstance from "../utils/axiosInstance";

const API_URL = "/articles"; // L'URL de base pour les endpoints des articles

export const getAllArticles = async () => {
  try {
    const response = await axiosInstance.get(API_URL);
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const getArticleById = async (articleId) => {
  try {
    const response = await axiosInstance.get(`${API_URL}/${articleId}`);
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const createArticle = async (userId, categoryIds, articleRequestDTO) => {
  try {
    const response = await axiosInstance.post(
      `${API_URL}/${userId}`,
      articleRequestDTO,
      {
        params: { categoryIds },
      }
    );
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const deleteArticle = async (articleId) => {
  try {
    await axiosInstance.delete(`${API_URL}/${articleId}`);
  } catch (error) {
    throw error;
  }
};

export const addFavorite = async (userId, articleId) => {
  try {
    await axiosInstance.post(`${API_URL}/${userId}/favorites/${articleId}`);
  } catch (error) {
    throw error;
  }
};

export const removeFavorite = async (userId, articleId) => {
  try {
    await axiosInstance.delete(`${API_URL}/${userId}/favorites/${articleId}`);
  } catch (error) {
    throw error;
  }
};

export const getFavoriteArticlesByUser = async (userId) => {
  try {
    const response = await axiosInstance.get(`${API_URL}/${userId}/favorites`);
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const getAllArticlesPaged = async (page, size, sort, order) => {
  try {
    const response = await axiosInstance.get(`${API_URL}/paged`, {
      params: { page, size, sort, order },
    });
    return response.data;
  } catch (error) {
    throw error;
  }
};
